pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -369,3 +369,56 @@ func avgHistograms(head, tail []promql.HPoint) (*histogram.FloatHistogram, error
 
 	return avgSoFar, nil
 }
+
+var Resets = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction:         DropSeriesName,
+	StepFunc:                       resets,
+	NeedsSeriesNamesForAnnotations: true,
+}
+
+func resets(step types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	fHead, fTail := step.Floats.UnsafePoints(step.RangeEnd)
+	hHead, hTail := step.Histograms.UnsafePoints(step.RangeEnd)
+
+	haveFloats := len(fHead) > 0 || len(fTail) > 0
+	haveHistograms := len(hHead) > 0 || len(hTail) > 0
+
+	if !haveFloats && !haveHistograms {
+		return 0, false, nil, nil
+	}
+
+	resets := 0.0
+
+	if haveFloats {
+		prev := fHead[0].F
+		accumulate := func(points []promql.FPoint) {
+			for _, sample := range points {
+				current := sample.F
+				if current < prev {
+					resets++
+				}
+				prev = current
+			}
+
+		}
+		accumulate(fHead)
```

Comment:
We should be able to skip the first element (same comment applies below for histogram case too):

```suggestion
		accumulate(fHead[1:])
```